### PR TITLE
Feat/catalan number

### PR DESCRIPTION
**Pull request by i-redbyte**

### CODE REVIEW

File: math/catalan/catalannumber.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// catalannumber.go
```
Comment (raklaptudirm) [inline math/catalan/catalannumber.go:1]: Add a small explanation of what is a Catalan number.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: math/catalan/catalannumber.go (hunk 1 above)
Comment (i-redbyte) [reply math/catalan/catalannumber.go:1]: Done

**Review by raklaptudirm — DISMISSED** [review]

File: math/catalan/catalannumber_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import "testing"
+
+func TestCatalanNumber(t *testing.T) {
+	type args struct {
+		n int
+	}
+	tests := []struct {
+		name string
+		args args
+		want int
```
Comment (tjgurwara99) [inline math/catalan/catalannumber_test.go:17]: There is only one argument right? Why did you make a struct to contain n when you can just have this?

```suggestion
	tests := []struct {
		name string
		arg  int
		want int
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: math/catalan/catalannumber_test.go (hunk 2 above)
Comment (i-redbyte) [reply math/catalan/catalannumber_test.go]: Done

**Review by tjgurwara99 — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**